google/cloud/pubsub/internal/publisher_tracing_connection_test.cc

```diff
@@ -15,35 +15,132 @@
 #ifdef GOOGLE_CLOUD_CPP_HAVE_OPENTELEMETRY
 
 #include "google/cloud/pubsub/internal/publisher_tracing_connection.h"
+#include "google/cloud/pubsub/message.h"
 #include "google/cloud/pubsub/mocks/mock_publisher_connection.h"
 #include "google/cloud/pubsub/publisher_connection.h"
-#include "google/cloud/internal/opentelemetry.h"
-#include "google/cloud/opentelemetry_options.h"
+#include "google/cloud/pubsub/topic.h"
+#include "google/cloud/future_generic.h"
+#include "google/cloud/internal/make_status.h"
+#include "google/cloud/status.h"
+#include "google/cloud/status_or.h"
 #include "google/cloud/testing_util/opentelemetry_matchers.h"
+#include "google/cloud/testing_util/status_matchers.h"
+#include "gtest/gtest.h"
 #include <gmock/gmock.h>
+#include <opentelemetry/trace/semantic_conventions.h>
 
 namespace google {
 namespace cloud {
 namespace pubsub_internal {
 namespace {
 
+auto constexpr kErrorCode = static_cast<int>(StatusCode::kAborted);
+
 using ::google::cloud::pubsub::PublisherConnection;
+using ::google::cloud::pubsub::Topic;
 using ::google::cloud::pubsub_internal::MakePublisherTracingConnection;
 using ::google::cloud::pubsub_mocks::MockPublisherConnection;
 using ::google::cloud::testing_util::InstallSpanCatcher;
 using ::google::cloud::testing_util::OTelAttribute;
 using ::google::cloud::testing_util::SpanHasAttributes;
 using ::google::cloud::testing_util::SpanHasInstrumentationScope;
 using ::google::cloud::testing_util::SpanKindIsClient;
+using ::google::cloud::testing_util::SpanKindIsProducer;
 using ::google::cloud::testing_util::SpanNamed;
 using ::google::cloud::testing_util::SpanWithStatus;
+using ::google::cloud::testing_util::StatusIs;
 using ::google::cloud::testing_util::ThereIsAnActiveSpan;
+using ::testing::AllOf;
+using ::testing::ElementsAre;
+
+TEST(PublisherTracingConnectionTest, PublishSpanOnSuccess) {
+  namespace sc = ::opentelemetry::trace::SemanticConventions;
+  auto span_catcher = InstallSpanCatcher();
+  auto mock = std::make_shared<MockPublisherConnection>();
+  EXPECT_CALL(*mock, Publish)
+      .WillOnce([&](pubsub::PublisherConnection::PublishParams const&) {
+        EXPECT_TRUE(ThereIsAnActiveSpan());
+        return make_ready_future(StatusOr<std::string>("test-id-0"));
+      });
+  auto connection = MakePublisherTracingConnection(
+      Topic("test-project", "test-topic"), std::move(mock));
+
+  auto response = connection
+                      ->Publish({pubsub::MessageBuilder{}
+                                     .SetData("test-data-0")
+                                     .SetOrderingKey("ordering-key-0")
+                                     .Build()})
+                      .get();
+
+  EXPECT_THAT(response, StatusIs(StatusCode::kOk));
```

Comment:
nit: I think we tend to say `EXPECT_STATUS_OK(response)`

Reply:
Done